snuba/datasets/eventsv2.py

```diff
@@ -0,0 +1,75 @@
+from datetime import timedelta
+from typing import Sequence
+
+from snuba.datasets import TimeSeriesDataset
+from snuba.datasets.dataset_schemas import DatasetSchemas
+from snuba.datasets.factory import get_dataset
+from snuba.datasets.schemas.join import (
+    JoinConditionExpression,
+    JoinCondition,
+    JoinedSchema,
+    JoinStructure,
+    JoinType,
+    SchemaJoinedSource,
+    SubJoinSource,
+)
+from snuba.query.extensions import PERFORMANCE_EXTENSION_SCHEMA, PROJECT_EXTENSION_SCHEMA
+from snuba.query.schema import GENERIC_QUERY_SCHEMA
+from snuba.request import RequestSchema
+from snuba.schemas import get_time_series_extension_properties
+
+
+class EventsV2(TimeSeriesDataset):
+    def __init__(self) -> None:
+        grouped_message = get_dataset("groupedmessage")
+        events = get_dataset("events")
+
+        join_structure = JoinStructure(
+            left_source=SchemaJoinedSource(
+                "groups",
+                grouped_message.get_dataset_schemas().get_read_schema(),
+            ),
+            right_source=SchemaJoinedSource(
+                "events",
+                events.get_dataset_schemas().get_read_schema(),
+            ),
+            mapping=[
+                JoinCondition(
+                    left=JoinConditionExpression(table_alias="groups", column="project_id"),
+                    right=JoinConditionExpression(table_alias="events", column="project_id"),
+                ),
+                JoinCondition(
+                    left=JoinConditionExpression(table_alias="groups", column="id"),
+                    right=JoinConditionExpression(table_alias="events", column="group_id"),
+                ),
+            ],
+            join_type=JoinType.LEFT,
+        )
+
+        schema = JoinedSchema(join_structure)
+        dataset_schemas = DatasetSchemas(
+            read_schema=schema,
+            write_schema=None,
+        )
+        super().__init__(
+            dataset_schemas=dataset_schemas,
+            processor=None,
+            default_topic=None,
+            time_group_columns={
+                'events.time': 'events.timestamp',
```

Comment:
I am hesitating whether we should refer to columns within a join as string with an alias inside (which is consistent with the way we address columns today. Except for when we build a schema, they are just strings). Or to build an abstraction on them, but that has big implications on everything that access the columns of a dataset.

Reply:
I changed it to keep the structure in the ColumnSet. 
Anyway this list (besides DDL which is irrelevant here) is used only to find if a column passed in the query (thus unstructured) is in the columnSet.
We will be able to do better when we will change the way we parse our query and once we will give more structure to the query object.